homeassistant/components/device_tracker/bluetooth_tracker.py

```diff
@@ -0,0 +1,89 @@
+"""Tracking for bluetooth devices."""
+import logging
+from datetime import timedelta
+
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.components.device_tracker import (
+    YAML_DEVICES,
+    CONF_TRACK_NEW,
+    CONF_SCAN_INTERVAL,
+    DEFAULT_SCAN_INTERVAL,
+    load_config,
+)
+import homeassistant.util as util
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pybluez==0.22']
+
+
+def setup_scanner(hass, config, see):
+    """Setup the Bluetooth Scanner."""
+    # pylint: disable=import-error
+    import bluetooth
+
+    def see_device(device):
+        """Mark a device as seen."""
+        see(dev_id='bt_' + device[1], mac=device[0], host_name=device[1])
```

Comment:
Why are you passing in device id ? It's not going to be used except for defining a key in `known_devices.yaml` which cannot be trusted - people can modify that one as they see fit because it defines how the entity id for the device tracker will be formed.

Instead, just pass `mac` and `hostname` as you do now and just prefix the mac with `bt_` as that's a value that users will not change. That way you will not have to update `device_tracker.__init__` and make finding bluetooth devices a lot easier.